File: BlockchainSimulation2/BlockchainSimulation2/Dtos/ClientRequestDto.cs
using System;
using System.Collections.Generic;
using BlockchainSimulation2.Database;

namespace BlockchainSimulation2.Dtos
{
    public class ClientRequestDto
    {
        public string Hash { get; set; }
        public ClientType Type { get; set; }
        public string Amount { get; set; }
        public DateTime StartDate { get; set; }

        public IEnumerable<string> TransactionsHashes { get; set; }
        public IEnumerable<string> MinedBlocksHashes { get; set; }
    }
}

File: BlockchainAgents/BlockchainAgents/src/main/java/blockchain/dto/ClientRequestDto.java
package blockchain.dto;

import blockchain.ethereumj.BasicNode;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class ClientRequestDto {
    public String Hash;
    public ClientType Type;
    public BigDecimal Amount;
    public Date StartDate;

    public List<String> TransactionsHashes;
    public List<String> MinedBlocksHashes;

    public ClientRequestDto() {
        TransactionsHashes = new ArrayList<>();
        MinedBlocksHashes = new ArrayList<>();
    }

    public static ClientRequestDto of(BasicNode node, boolean isMiner) {
        ClientRequestDto dto = new ClientRequestDto();

        dto.Hash = node.getAddressString();
        dto.Amount = node.getBalance();
        dto.StartDate = new Date(System.currentTimeMillis());
        dto.Type = isMiner ? ClientType.Miner : ClientType.Client;

        return dto;
    }
}
